uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -656,13 +670,14 @@ define([
             initContact.call(self);
 
             if (self.isMode('edit')) {
-              initialLeaveRequestAttributes = self.request.attributes();
+              initialLeaveRequestAttributes = angular.copy(self.request.attributes());
+              // fileUploader property deleted because it will not be used
+              // in object comparison
+              delete initialLeaveRequestAttributes.fileUploader;
```

Comment:
`initialLeaveRequestAttributes` is stored only once, so if I do the removal of property in `hasRequestChanged` then unnecessarily we have to remove the property every time while its enough to remove it only once as the object is not changing. Also `hasRequestChanged` is a scope function, so it gets called numerous times, it will not be efficient to remove the property there.